### PR TITLE
Fix #131 - Validate email format in session

### LINKED ISSUE #131: Validate email format in Session
Using an EmailField.

### PR DESCRIPTION
**Pull request by flyusfly**
. Add a new EmailField type.
   . Update email type from CharField to EmailField.
   . Add 2 session tests (Email valid  & Email not valid test).

### CODE REVIEW

**Comment by flyusfly** [comment]
Let's discuss about it?

File: ban/tests/auth/test_session.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        'client': client.name
+    }
+
+
+def test_session_should_have_either_a_client_or_a_user():
+    with pytest.raises(ValueError):
+        models.Session.create()
+
+
+def test_session_can_be_created_with_a_user_and_a_valid_Email():
```
Comment (vinyll) [inline ban/tests/auth/test_session.py]: lower case 'Email'

File: ban/tests/auth/test_session.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    user = UserFactory()
+    session = models.Session.create(user=user, email='foo@foo.com')
+    assert session.email == 'foo@foo.com'
+
+
+def test_session_can_not_be_created_with_a_user_and_an_unvalid_Email():
```
Comment (vinyll) [inline ban/tests/auth/test_session.py]: lower case 'Email'

File: ban/tests/auth/test_session.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    session = models.Session.create(client=client)
+    assert session.client == client
+    assert session.as_relation == {
+        'id': session.pk,
+        'user': None,
+        'client': client.name
+    }
+
+
+def test_session_should_have_either_a_client_or_a_user():
```
Comment (vinyll) [inline ban/tests/auth/test_session.py]: What is this testing?

File: ban/db/fields.py [hunk 4]
```diff
@@ -251,3 +253,16 @@ def python_value(self, value):
...
+
+class EmailField(CharField):
+
+    max_length = 100
+
+    def coerce(self, value=None):
+        if value:
+            value = str(value)
+            if not validate_email(value):
+                raise ValueError('Invalid Email "{}"'.format(value))
```
Comment (vinyll) [inline ban/db/fields.py]: lower case 'Email'

File: ban/db/fields.py [hunk 5]
```diff
@@ -251,3 +253,16 @@ def python_value(self, value):
         if value is None:
             return value
         return super().python_value(value)
+
+
+class EmailField(CharField):
+
+    max_length = 100
+
+    def coerce(self, value=None):
```
Comment (vinyll) [inline ban/db/fields.py]: I would remove an indent with the negate check first:
```python
if not value:
    return None
value = str(value)
if not validate_email(value):
    raise ValueErr…
```

File: ban/db/fields.py [hunk 6]
```diff
@@ -11,7 +12,8 @@
 __all__ = ['PointField', 'ForeignKeyField', 'CharField', 'IntegerField',
            'HStoreField', 'UUIDField', 'ArrayField', 'DateTimeField',
            'BooleanField', 'BinaryJSONField', 'PostCodeField', 'FantoirField',
-           'ManyToManyField', 'PasswordField', 'DateRangeField', 'TextField']
+           'EmailField', 'ManyToManyField', 'PasswordField', 'DateRangeField',
+           'TextField']
 
 
 lonlat_pattern = re.compile('^[\[\(]{1}(?P<lon>-?\d{,3}(:?\.\d*)?), ?(?P<lat>-?\d{,3}(\.\d*)?)[\]\)]{1}$')  # noqa
```
Comment (vinyll) [inline ban/db/fields.py]: Maybe can you split it to make it length valid without a _noqa_:
```python
lonlat_pattern = re.compile('^[\[\(]{1}(?P<lon>-?\d{,3}(:?\.\d*)?),'
                             ' ?(?P<lat>-?\d{,3}(\.\d*)?)[\]\)]{1}$')
```

File: ban/tests/auth/test_session.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+from ban.tests.factories import UserFactory, ClientFactory
+
+
+def test_session_can_be_created_with_a_user():
+    user = UserFactory()
+    session = models.Session.create(user=user)
+    assert session.user == user
+    assert session.as_relation == {
+        'id': session.pk,
+        'user': user.username,
```
Comment (vinyll) [inline ban/tests/auth/test_session.py]: Do we actually want to store the _username_ and not the pk?

File: ban/tests/auth/test_session.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+
+def test_session_can_be_created_with_a_client():
+    client = ClientFactory()
+    session = models.Session.create(client=client)
+    assert session.client == client
+    assert session.as_relation == {
+        'id': session.pk,
+        'user': None,
+        'client': client.name
```
Comment (vinyll) [inline ban/tests/auth/test_session.py]: Do we actually want to store the _client's name_ and not the pk?

**Pull request closed without merging**